tests/unit/test_speechManager/__init__.py

```diff
@@ -1110,3 +1112,28 @@ class InitialDevelopmentTests_withCancellableSpeechEnabled(InitialDevelopmentTes
 	def setUp(self):
 		super().setUp()
 		config.conf['featureFlag']['cancelExpiredFocusSpeech'] = 1  # yes
+
+
+class Test_pr11651(unittest.TestCase):
+
+	def test_redundantSequenceAfterEndUtterance(self):
+		"""
+		Tests that redundant param change and index commands are not emitted as an extra utterance
+		when the preceeding utterance contained param change commands and an EndUtterance command.
+		E.g. speaking a character.
+		"""
+		smi = SpeechManagerInteractions(self)
+		seq = [
+			CharacterModeCommand(True),
+			"a",
+			smi.create_ExpectedIndex(1),
+			EndUtteranceCommand(),
+		]
+		with smi.expectation():
+			smi.speak(seq)
+			# synth should receive the characterMode, the letter 'a' and an index command.
+			smi.expect_synthSpeak(sequence=seq[:-1])
+		with smi.expectation():

```

Comment:
Perhaps add a comment here to show what this previously did, just to clarify how it was wrong:
```suggestion
		with smi.expectation():
			# Previously, this would result in synth.speak receiving
			# a call with sequence:
			# [CharacterModeCommand(True), IndexCommand(2)]
			# This is a problem because it includes an index command but no speech.
			# This is inefficient, and also some SAPI5 synths such as Ivona will not
			# notify of this bookmark.
```